### ISSUE TITLE
SC2016 false positive when using exec

### ISSUE THREAD

**Issue by kevinoid** [koalaman/shellcheck#2008]
#### For bugs
- Rule Id (if any, e.g. SC1000): SC2016
- My shellcheck version (`shellcheck --version` or "online"): online
- [x] The rule's wiki page does not already cover this (e.g. https://shellcheck.net/wiki/SC2086)
- [x] I tried on https://www.shellcheck.net/ and verified that this is still a problem on the latest commit

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/bin/sh
exec jq -n --arg s sval '$s'
```

#### Here's what shellcheck currently says:

    In /tmp/foo.sh line 2:
    exec jq -n --arg s sval '$s'
                            ^--^ SC2016: Expressions don't expand in single quotes, use double quotes for that.

    For more information:
      https://www.shellcheck.net/wiki/SC2016 -- Expressions don't expand in singl...

#### Here's what I wanted or expected to see:

No errors, which is what I see if `exec` is removed.

**Comment by koalaman** [comment]
Handling `exec` as well as `builtin` and `command` has now been improved. Thanks!

**Comment by kevinoid** [comment]
Looks great.  Thanks @koalaman!
